Source/Core/Common/TitleDatabase.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <fstream>
+#include <iostream>
+
+#include "Common/FileUtil.h"
+#include "Common/StringUtil.h"
+#include "Common/TitleDatabase.h"
+
+namespace Common
+{
+TitleDatabase::TitleDatabase()
+{
+  // Titles that cannot be part of the Wii TDB,
+  // but common enough to justify having entries for them.
+  m_title_map.emplace("0000000100000002", "System Menu");
+  m_title_map.emplace("OHBC", "The Homebrew Channel");
+
+  const std::string& load_directory = File::GetUserPath(D_LOAD_IDX);
+  Load(File::GetSysDirectory() + "wiitdb.txt");
+  if (!Load(load_directory + "wiitdb.txt"))
+    Load(load_directory + "titles.txt");
+}
+
+TitleDatabase::~TitleDatabase() = default;
+
+bool TitleDatabase::Load(const std::string& file_path)
+{
+  std::ifstream txt;
+  OpenFStream(txt, file_path, std::ios::in);
+
+  if (!txt.is_open())
+    return false;
+
+  std::string line;
+  while (std::getline(txt, line))
+  {
+    if (line.empty())
+      continue;
+
+    const size_t equals_index = line.find('=');
+    if (equals_index != std::string::npos)
+    {
+      m_title_map.emplace(StripSpaces(line.substr(0, equals_index)),
+                          StripSpaces(line.substr(equals_index + 1)));
+    }
+  }
+  return true;
+}
+
+std::string TitleDatabase::GetTitleName(const std::string& game_id) const
+{
+  auto iterator = m_title_map.find(game_id);
+
+  // The title database may not have entries for the full game ID (6 characters),
+  // so we should also search for a matching ID with only four characters.
```

Comment:
This approach makes it impossible to distinguish between e.g. Mario Kart Wii and the Mario Kart Channel. (See PR #2242.) There needs to be a parameter to this function that says whether the title is a disc or a channel.